deploy/hpilo.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env sh
+
+# Deploy script for HPE iLO4
+#
+# The following environment variables are
+# needed for the deploy script to work:
+#
+# ```sh
+# export HPILO_USERNAME=admin
+# export HPILO_PASSWORD=secret
+# export HPILO_HOST=ilo.example.com
+#
+# acme.sh --deploy -d ilo.example.com --deploy-hook hpilo
+# ```
+
+########  Public functions #####################
+
+#domain keyfile certfile cafile fullchain
+hpilo_deploy() {
+  _cdomain="$1"
+  _ckey="$2"
+  _ccert="$3"
+  _cca="$4"
+  _cfullchain="$5"
+
+  if [ -f "$DOMAIN_CONF" ]; then
+    # shellcheck disable=SC1090
+    . "$DOMAIN_CONF"
+  fi
+
+  _debug _cdomain "$_cdomain"
+  _debug _ckey "$_ckey"
+  _debug _ccert "$_ccert"
+  _debug _cca "$_cca"
+  _debug _cfullchain "$_cfullchain"
+
+  # iLO host is optional, use _cdomain if not provided
+  if [ -n "$HPILO_HOST" ]; then
```

Comment:
you don't need 2 names: `HPILO_HOST` and `Le_Deploy_ilo_host`

You should use one name like: `DEPLOY_HPILO_HOST`,  and use `_getdeployconf()` and `_savedeployconf`

see example:

https://github.com/acmesh-official/acme.sh/blob/master/deploy/docker.sh#L29